tests/test_ase.py

```
@@ -0,0 +1,290 @@
+        # Component 2: atoms 2 and 3 connected
+        bond = [[1], [0], [3], [2]]  # 0-1 and 2-3
+        level = [[1], [1], [1], [1]]
+
+        # Use the _connectmolecule method which should use connected_components
+        result = detect_class._connectmolecule(bond, level)
+
+        # Should have 2 components
+        assert len(result) == 2
+
+        # Each component should have 2 atoms
+        for component in result:
```

Ruff (0.14.10)

Loop control variable `component` not used within loop body

Rename unused `component` to `_component`

(B007)